pkgs/applications/misc/blender/derivation.nix

```
@@ -0,0 +1,280 @@
+{ config, stdenv, lib, fetchFromGitea, fetchzip, boost, cmake, ffmpeg, gettext, glew
+, ilmbase, libepoxy, libXi, libX11, libXext, libXrender
+, libjpeg, libpng, libsamplerate, libsndfile
+, libtiff, libwebp, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio, openjpeg
+, openvdb, libXxf86vm, tbb, alembic
+, zlib, zstd, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
+, jackaudioSupport, libjack2
+, cudaSupport, cudaPackages ? { }
+, hipSupport, rocmPackages # comes with a significantly larger closure size
+, colladaSupport, opencollada
+, spaceNavSupport, libspnav
+, makeWrapper
+, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
+, waylandSupport, pkg-config, wayland, wayland-protocols, libffi, libdecor, libxkbcommon, dbus
+, potrace
+, openxr-loader
+, embree, gmp, libharu
+, openpgl
+, mesa
+, runCommand
+, version, hashes, isLTS, pythonPackages, withPackages, xvfb-run
+}:
+
+let
+  python = pythonPackages.python;
+  optix = fetchzip {
+    # url taken from the archlinux blender PKGBUILD
+    url = "https://developer.download.nvidia.com/redist/optix/v7.3/OptiX-7.3.0-Include.zip";
+    sha256 = "0max1j4822mchj0xpz9lqzh91zkmvsn4py0r174cvqfz8z8ykjk8";
+  };
+  blender-addons = fetchFromGitea {
+    domain = "projects.blender.org";
+    owner = "blender";
+    repo = "blender-addons";
+    rev = "v${version}";
+    hash = hashes.addons;
+
+    postFetch = "patch -p3 -d $out < ${./draco-p2.patch}";
+  };
+
+in
+stdenv.mkDerivation (finalAttrs: rec {
+  inherit version;
+
+  pname = "blender";
+
+  src = fetchFromGitea {
+    domain = "projects.blender.org";
+    owner = "blender";
+    repo = "blender";
+    rev = "v${version}";
+    hash = hashes.blender;
+
+    # Blender 3.3 uses submodules, but 3.5+ doesnt.
+    # See https://projects.blender.org/blender/blender/pulls/104755
+    fetchSubmodules = if lib.versionOlder version "3.5" then true else false;
+  };
+
+  prePatch =
+    if lib.versionAtLeast version "3.5" then ''
+      mkdir scripts/addons
+      cp -Rv ${blender-addons}/* scripts/addons
+    '' else ''
+      cp -Rv ${blender-addons}/* release/scripts/addons
+    '';
+
+  patches = [
+    ./draco-p1.patch
+  ] ++ lib.optional stdenv.isDarwin ./darwin.patch;
+
+  nativeBuildInputs =
+    [ cmake makeWrapper pythonPackages.wrapPython llvmPackages.llvm.dev
+    ]
+    ++ lib.optionals cudaSupport [ addOpenGLRunpath ]
```

```suggestion
  nativeBuildInputs = [
    cmake
    makeWrapper
    pythonPackages.wrapPython
    llvmPackages.llvm.dev
  ] ++ lib.optionals cudaSupport [ addOpenGLRunpath ]
```
if we are refactoring we might as well use consistent one per line style